core/src/main/scala/com/metamx/tranquility/beam/Beam.scala

```diff
@@ -19,22 +19,41 @@
 package com.metamx.tranquility.beam
 
 import com.twitter.util.Future
+import scala.collection.immutable.BitSet
+import scala.collection.mutable
 
 /**
- * Beams can accept events and forward them along. The propagate method may throw a DefunctBeamException, which means
- * the beam should be discarded (after calling close()).
- */
-trait Beam[A]
+  * Beams can accept messages and forward them along. The propagate method may throw a DefunctBeamException, which
+  * means the beam should be discarded (after calling close()).
+  */
+trait Beam[MessageType]
 {
   /**
-   * Request propagation of events. The operation may fail in various ways, which tend to be specific to
-   * the implementing class. Returns the number of events propagated.
-   */
-  def propagate(events: Seq[A]): Future[Int]
+    * Request propagation of messages. The return value indicates the number of messages known to be sent
+    * successfully. Note that for some implementations, it is possible for a message to be sent and for the ack to
+    * be lost (e.g. if the send is occurring over a network).
+    * @param messages a batch of messages
+    * @return the number of messages propagated
+    */
+  @deprecated("use sendBatch", "0.7.0")
+  final def propagate(messages: Seq[MessageType]): Future[Int] = {
+    sendBatch(messages).map(_.size)
+  }
+
+  /**
+    * Request propagation of messages. The returned bitset contains the indexes of messages known to be sent
+    * successfully. Note that for some implementations, it is possible for a message to be sent and for the ack to be
+    * lost (e.g. if the send is occurring over a network).
+    * @param messages a batch of messages
+    * @return a bitset containing indexes of messages that were sent successfully
+    */
+  def sendBatch(messages: Seq[MessageType]): Future[BitSet]
```

Comment:
didn't see your comment in the prior comment thread. Since this is not a new interface I don't consider it a blocker, but in general if a sequence is being indexed, it would make sense to me that the passed index must be immutable.

I'm OK as is simply to maintain compat with prior behavior.